### PR TITLE
GIPF-87 game turns

### PR DESCRIPTION
**Pull request by sulphite**
- [x] on joining a room you are assigned a player colour
- [x] black plays first, and control switches to the other player after their move
- [x] pieces placed are of the correct colour
- [x] players cannot interact with the board unless it is their turn

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -30,33 +32,44 @@ function App() {
       switch (messageData.type) {
         // on joining room, room and player colour are set
         // board is loaded
+        // set current player to true
         case "roomJoined": {
+          //console.log(messageData.data)
           setRoom(messageData.data.room)
-          setColour(messageData.data.playerColour)
+          setInitialColour(messageData.data.playerColour)
+          console.log("playercolour from backend",messageData.data.playerColour)
```
Comment (adrianHards) [inline frontend/src/App.tsx:40]: is this needed?

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -65,7 +78,7 @@ function App() {
 
         //
         default:
-          console.log(messageData)
+          console.log("unknown message type", messageData)
```
Comment (adrianHards) [inline frontend/src/App.tsx:81]: is this needed?

**Review by adrianHards — APPROVED** [review]
see comments

File: frontend/src/App.tsx (hunk 2 above)
Comment (sulphite) [reply frontend/src/App.tsx:81]: Its useful to see if the messages come through wrong

File: frontend/src/App.tsx (hunk 1 above)
Comment (sulphite) [reply frontend/src/App.tsx:40]: not really

**Pull request merged**